### PR TITLE
✨Adds reserved disk space to dynamic-sidecar

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

When the dynamic-sidecar starts it now reserves a block of 10MiB from the disk to be freed in case of emergencies.

If the underlying hosts runs out of disk space, this comes in handy.

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_scheduler.py [hunk 1]
```diff
@@ -256,3 +256,17 @@ async def _task_cleanup_service_docker_resources(
         )
     except TaskAlreadyRunningError as e:
         raise HTTPException(status.HTTP_409_CONFLICT, detail=f"{e}") from e
+
+
+@router.post(
+    "/services/{node_uuid}/disk/emergency:free",
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_scheduler.py:262]: Q: is emergency a resource now?

**Review by sanderegg — COMMENTED** [review]
Ok I like the idea. You reserve some space that is always available for the preferences. That is very nice.

Now I have a few suggestions/questions here:
- can you maybe rename this space from "emergency" to "reserved" ?
- would it not be possible to automatically free it just before saving the preferences instead of all that manual stuff that I guess you will need to do if this happens?
- alternatively, why not have the preferences with the current values/defaults saved on start so that the space is already occupied and just needs to be rewritten?
- I guess this would also include possible screenshots? then the space might not be sufficient.

File: services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_scheduler.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_scheduler.py:262]: So the idea is that the `reserved`(the new name) is a sub-resource of disk which is also a resource. Then this one gets freed `"/services/{node_uuid}/disk/reserved:free"`

**Comment by GitHK** [comment]
> Ok I like the idea. You reserve some space that is always available for the preferences. That is very nice.
> 
> Now I have a few suggestions/questions here:
> 
> * can you maybe rename this space from "emergency" to "reserved" ?

✅ 

> * would it not be possible to automatically free it just before saving the preferences instead of all that manual stuff that I guess you will need to do if this happens?

good point, did not think about it. will still leave in the functionality to remove it manually (since stuff can always go wrong)

> * alternatively, why not have the preferences with the current values/defaults saved on start so that the space is already occupied and just needs to be rewritten?
> * I guess this would also include possible screenshots? then the space might not be sufficient.

I did not understand these last two points

**Review by sanderegg — APPROVED** [review]
thank you!

**Pull request merged**